src/sails_forth/memory.clj

```diff
@@ -274,3 +274,11 @@
 (defn limits!
   [astate]
   {})
+
+(defn take-action!
+  [astate take-action-map action inputs]
+  (let [f (get take-action-map action)]
+    (if f
+      (f astate inputs)
+      (throw (ex-info "action failed" {:cause (str f "not implemented")
```

Comment:
shouldn't this be reporting the endpoint/action that's not implemented? since we've just established `f` is nil/false